### ISSUE TITLE
Radio button not accessible in High Contrast

### ISSUE THREAD

**Issue by MarcelPoirier** [carbon-design-system/carbon#7280]
When in high contrast you can't tell which radio button is selected.

## Environment
Window in High Contrast
Firefox ESR 78

## Detailed description
User should be able to tell which radio button is selected

> What WCAG 2.1 checkpoint does the issue violate?
Success Criterion 1.4.11 Non-text Contrast https://www.w3.org/TR/WCAG21/#non-text-contrast

## Steps to reproduce the issue

1. Windows set to High Contrast
2. Visit https://www.carbondesignsystem.com/components/radio-button/usage with Firefox
3. Scroll down to the example radio button
4. Bug - can't see which on is selected
![image](https://user-images.githubusercontent.com/31481145/98957540-4a669880-24cf-11eb-9907-4146777de60e.png)

**Comment by tw15egan** [comment]
Which version of the components are you on? I believe a fix was added recently to fix this

https://github.com/carbon-design-system/carbon/pull/6892

https://github.com/carbon-design-system/carbon/pull/6892/files#diff-48f235115e4527b3813176722ad79ba66e96fdc740656af9f77e3217ff3cbafaR101-R107

Let me know if after updating you still see these issues and I'll take a look 👍

**Comment by MarcelPoirier** [comment]
We're on "carbon-components-react": "7.17.0".  You're right, looks like that change fixes the issue.  Thank you.

**Comment by emyarod** [comment]
I thought this might have been a regression since this is what I see in Firefox in HCM

![image](https://user-images.githubusercontent.com/8265238/98986905-1f8a3d80-24eb-11eb-9a4f-d6eb580bdb36.png)

but I think I'm just missing a browser option

**Comment by MarcelPoirier** [comment]
Note, I haven't tested with the latest Carbon build.  I simply applied the fix locally to make sure it worked.  Switching Carbon versions in our product isn't straight forward.

**Comment by tw15egan** [comment]
@emyarod FYI this is what I had to do to enable it in Firefox:

![image (1)](https://user-images.githubusercontent.com/11928039/99004123-8b70a400-24f3-11eb-9f06-9368b25b4e90.png)

**Comment by emyarod** [comment]
yep that is the browser option I was referring to

**Comment by tw15egan** [comment]
Going to close as resolved, feel free to reopen @MarcelPoirier if there is any other action needed on our end 👍

**Comment by angella-qian** [comment]
@tw15egan Wasn't sure if I should open a separate thread but I still see this bug on Windows High Contrast Mode using Google Chrome. The radio buttons aren't showing a selected state even as I click them. I do see them working fine on Edge, though.
<img width="197" alt="Screen Shot 2020-11-30 at 11 38 51 AM" src="https://user-images.githubusercontent.com/36805013/100779701-c1a89180-33d6-11eb-8f18-b516221c279e.png">

**Comment by tw15egan** [comment]
@angella-qian Hey there! Unfortunately, Google Chrome does not provide proper OS-level HCM support, therefore we are only able to provide proper HCM in Windows 10 on Edge/IE11. Firefox has recently added a `prefers-contrast` flag in v80+ that we've been able to tap into, so we've started utilizing that as well, but it is still experimental and we've encountered some bugs with it. I'd recommend testing HCM issues on Windows 10 in Edge and see if they are resolved 👍

**Comment by angella-qian** [comment]
@tw15egan Thanks for the quick response! So I just tested on our dev slot containing the latest carbon (`"carbon-components-react": "^7.24.0",`) on Firefox and looks like the radio buttons still aren't showing their selected states on our page. It's weird because I see them working fine on https://www.carbondesignsystem.com/components/radio-button/usage/
<img width="159" alt="Screen Shot 2020-12-01 at 11 35 40 AM" src="https://user-images.githubusercontent.com/36805013/100788143-ad6a9180-33e2-11eb-8f0b-aada0a3000e3.png">

I checked again and can verify they're working on Edge
<img width="166" alt="Screen Shot 2020-12-01 at 11 33 51 AM" src="https://user-images.githubusercontent.com/36805013/100787982-72685e00-33e2-11eb-944c-c5b89ba8631e.png">

**Comment by tw15egan** [comment]
To enable `prefers-contrast`, you'll need to make sure you're on Firefox v80+, type `about:config` into the address bar, and then search for `prefers contrast` and set that to true. Like I said this is still an experimental feature from Mozilla so its not fully supported, and any HCM testing should be done in Edge

**Comment by tw15egan** [comment]
Increased Firefox support was added in a recent PR ( https://github.com/carbon-design-system/carbon/pull/7314) and should be included in `7.25.0` when it's released

**Comment by angella-qian** [comment]
Yeah I did enable that setting which is why I was able to see the radio buttons working on [this page](https://www.carbondesignsystem.com/components/radio-button/usage/ ). Just not sure why they're not working in our application. I'll relay this back to the team though, thanks for the updates
